controllers/tobiko_controller.go

```diff
@@ -288,3 +275,67 @@ func (r *TobikoReconciler) EnsureTobikoCloudsYAML(ctx context.Context, instance
 
 	return ctrl.Result{}, nil
 }
+
+// This function prepares env variables for a single workflow step:
+//   - It first sets the global configuration from the spec section
+//   - Then it updates the env variables based on the workflow step
+//     that should be executed.
+func (r *TobikoReconciler) PrepareTobikoEnvVars(ctx context.Context, instance *testv1beta1.Tobiko, helper *helper.Helper, step int) map[string]env.Setter {
+	// Prepare env vars
+	envVars := make(map[string]env.Setter)
+	envVars["TOBIKO_TESTENV"] = env.SetValue(r.OverwriteStringValueWithWorkflow(instance.Spec.Testenv, instance.Spec.Workflow[step].Testenv, "py3"))
+	envVars["TOBIKO_VERSION"] = env.SetValue(r.OverwriteStringValueWithWorkflow(instance.Spec.Version, instance.Spec.Workflow[step].Version, ""))
+	envVars["TOBIKO_PYTEST_ADDOPTS"] = env.SetValue(r.OverwriteStringValueWithWorkflow(instance.Spec.PytestAddopts, instance.Spec.Workflow[step].PytestAddopts, ""))
+	envVars["USE_EXTERNAL_FILES"] = env.SetValue("True")
+
+	if instance.Spec.PreventCreate {
```

Comment:
don't we need something like this here?
`r.OverwriteBoolValueWithWorkflow(instance.Spec.PreventCreate, instance.Spec.Workflow[step].PreventCreate, false)`

Reply:
True, thank you. I'm going to fix this.